deps/rabbitmq_mqtt/src/rabbit_mqtt_processor.erl

```diff
@@ -293,6 +295,20 @@ process_request(?PUBACK,
             {ok, State}
     end;
 
+%% MQTT 5 spec 3.3.1.2 QoS
+%% If the Server included a Maximum QoS in its CONNACK response
+%% to a Client and it receives a PUBLISH packet with a QoS greater than this
+%% then it uses DISCONNECT with Reason Code 0x9B (QoS not supported).
+process_request(?PUBLISH,
+                #mqtt_packet{fixed = #mqtt_packet_fixed{qos = ?QOS_2}},
+                State = #state{cfg = #cfg{
+                                        proto_ver = ?MQTT_PROTO_V5,
+                                        client_id = ClientID}}) ->
+    ?LOG_DEBUG("Terminating MQTT connection. QoS not supported, client ID: ~s ,"
+               "protocol version: ~p, qos: ~p",
+               [ClientID, ?MQTT_PROTO_V5, ?QOS_2]),
```

Comment:
Some suggestions on the log line:
1. Given this is a protocol error, should we increase the log level? In other places we log protocol errors at `error` level.
2. For logs output to the client, I would capitalise, i.e. writing `QoS` instead of `qos`.
3. I would first write the comma, then the space. So, instead of `client ID: myid ,protocol version`, it should be `client ID: my_id, protocol version`.